tools/check_ai_images.py

```diff
@@ -14,41 +15,27 @@
 logger = logging.getLogger(__name__)
 logging.getLogger("__main__").setLevel(logging.INFO)
 
+DEFAULT_REGISTRY = "quay.io/ocpmetal"
 
-def image_does_not_exist(image, tag):
-    cmd = f"docker manifest inspect quay.io/ocpmetal/{image}:{tag}"
-    try:
-        image_inspect_output = subprocess.check_output(cmd, shell=True)
-        image_inspect_output = subprocess.check_output("echo $?", shell=True)
-    except subprocess.SubprocessError:
-        image_inspect_output = "-1"
-
-    return int(image_inspect_output)
 
-
-def image_exist_with_retry(image, tag):
-    for retry in range(3):
-        get_image_result = image_does_not_exist(image, tag)
-        if not get_image_result:
-            return True
-        logger.warning(f"Failed to get image at {retry} retry")
-        time.sleep(10*60)
-
-    logger.error(f"Failed to get image at {retry} retry")
-    return not get_image_result
+@retry(exceptions=subprocess.SubprocessError, tries=3, delay=10)
+def does_image_exist(pull_spec: str) -> bool:
+    return subprocess.call(f"skopeo inspect --config docker://{pull_spec}", stdout=subprocess.DEVNULL, shell=True) == 0
```

Comment:
It's better not to use `shell=True` with variable input (`pull_spec`), because if it contains shell characters such as `\&{}` it'll cause trouble. Better to just `["skopeo", "inspect", "--config", "docker://{pull_spec}"]`

Reply:
yeah let's replace it
@omertuc you wanna do it real quick?

Reply:
https://github.com/openshift-assisted/assisted-installer-deployment/pull/133